src/components/Register.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react'
+import '../assets/styles/register.scss'
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import UserService from '../services/UserService';
+
+const SignupSchema = Yup.object().shape({
+    first_name: Yup.string()
+        .min(2, 'Too Short!')
+        .max(50, 'Too Long!')
+        .required('Required'),
+    last_name: Yup.string()
+        .min(2, 'Too Short!')
+        .max(50, 'Too Long!')
+        .required('Required'),
+    email: Yup.string().email('Invalid email').required('Required'),
+    password: Yup.string().min(6, 'You must enter 6 characters at least!').required('Password is required')
+});
+
+const Register = ({history}) => {
+    return (
+        <div>
+            <Formik
+                initialValues={{ email: '', password: '', first_name: '', last_name: '' }}
+                validate={values => {
+                    const errors = {};
+
+                    if (!values.email) {
+                        errors.email = 'Please enter email';
+                    }
+                    if (!values.password) {
+                        errors.password = 'Please enter password';
+                    }
+                    if (!values.first_name) {
+                        errors.first_name = 'Please enter first name';
+                    }
+                    if (!values.last_name) {
+                        errors.last_name = 'Please enter last name';
+                    }
+
+                    return errors;
+                }}
+                validationSchema={SignupSchema}
+                onSubmit={(values) => {
```

Comment:
You can create a separate function for handling form submission. 

```
const handleSubmit = values => {
   UserService.register(values);
}
```

and call that one from here. 

What is even better - you can use `useDispatch()` hook that is provided by `react-redux`. You will need to create action for registration (I can't see that one right now), and you can dispatch that one from `handleSubmit` function.

```
export const register = payload => {
    return {
        type: REGISTER_USER, // this is just a constant containing action type
        payload
    }
}
```
And in your component: 

`const dispatch = useDispatch();`
```
const handleSubmit = user => {
   dispatch(register(user));
}
```

For that, you will also need to create a saga that will be connected to a specific action, and handle this redirection to "/". This will prevent direct communication between components and services.